include/mrdocs/Generator.hpp

```diff
@@ -157,6 +157,23 @@ class MRDOCS_VISIBLE
     buildOneString(
         std::string& dest,
         Corpus const& corpus) const;
+
+    /** Prepare the output path for single-page documentation.
+
+        This function takes the specified `outputPath`, determines if it is a directory or file, 
+        and adjusts it to create the final output file path with the appropriate extension for 
+        the generator. If `outputPath` is a directory, this function appends "reference" as 
+        the filename and adds the generator's specific extension.
+
+        @param outputPath The original path, which can be a directory or a filename.
+
+        @return A string representing the complete path to the output file with 
+        the appropriate extension.
+    */
+    MRDOCS_DECL
+    std::string
+    prepareOutputPath(
```

Comment:
There's no reason for this to be a member function. It needs no members of the `Generator`, and not all generators will need it. 

If we need to reuse this logic, we can put that in Path.hpp with a name that makes sense (`prepare` is not informative), but I suspect we already have functions there to implement this logic with very few lines.

Reply:
I'll rename the function to `getOutputFullPath` to better reflect its purpose. Regarding making it a free function or moving it to Path, I believe it doesn't fit there. While not all generators use it, it's already used in `AdocGenerator`, `HtmlGenerator`, and the base `Generator`, making it closely tied to generator-specific logic.

As for the dependency on `fileExtension()`, I could make it a parameter to decouple it further, but since it's tied to the generator's behavior, I think keeping it as a member function maintains clarity.